spec/unit/relation/qualified_spec.rb

```diff
@@ -17,9 +17,21 @@
                     qualified.
                     group(:id)
 
-      expect(qualified.schema.all?(&:qualified)).to be(true)
+      expect(qualified.schema.all?(&:qualified?)).to be(true)
 
       expect(qualified.to_a).to eql([{ id: 1, count: 1 }, { id: 2, count: 1 }])
     end
+
+    it 'does not qualifies attribute without dataset' do
+      qualified = relation.select_append { `'hello'`.as(:bar) }.qualified
+
+      expect(qualified.schema.all?(&:qualified?)).to be(false)
+    end
+
+    it 'does not raise NoMethodError when qualifies attribute without dataset' do
+      expect {
+        relation.select_append { `'hello'`.as(:bar) }.qualified
+      }.to_not raise_error
+    end
```

Comment:
I wouldn't write a test that checks if it raises or not, a better way is to simply check expected behavior, ie:

``` ruby
qualified = relation.select_append { `'hello'`.as(:bar) }.qualified

expect(qualified.schema[:bar]).to_not be(nil) # or some better check?
```

Reply:
I guess by removing this spec and simply leaving the one on above should work right? https://github.com/rom-rb/rom-sql/pull/292/files#diff-f159d65edb9c79561a91c5f0b3c6f4e3R25

Reply:
@GustavoCaso yeah sounds good